package com.basic;
public class RethrowExceptionInCatchBlock {
    public static void main(String[] args) throws CustomException {
        try {
           
            throwException();
            
        } catch (CustomException e) {
        	
            // Additional handling or logging
            System.out.println("Caught CustomException: " + e.getMessage());

            // Rethrow the same exception
            throw e;
        }
    }

    private static void throwException() throws CustomException {
        
        throw new CustomException("An error occurred.");
    }
}

class CustomException extends Exception {
  
	private static final long serialVersionUID = 1L;

	public CustomException(String message) {
        super(message);
    }
}


